crates/wasmtime/src/module.rs

```diff
@@ -1042,6 +1042,47 @@ impl Module {
         self.inner.memory_images()?;
         Ok(())
     }
+
+    // XXX: Only for use within this repo's workspace.
+    #[doc(hidden)]
+    pub fn compiled_functions_info(&self) -> CompiledFunctionsInfo<'_> {
+        CompiledFunctionsInfo {
+            text: self.code_object().code_memory().text(),
+            func_locs: self
+                .compiled_module()
+                .finished_functions()
+                .map(|(f, _)| *self.compiled_module().func_loc(f))
+                .collect(),
+            address_map_section: self.code_object().code_memory().address_map_data(),
+        }
+    }
+}
+
+// XXX: Only for use within this repo's workspace.
+#[doc(hidden)]
+pub struct CompiledFunctionsInfo<'a> {
+    text: &'a [u8],
+    func_locs: Vec<FunctionLoc>,
+    address_map_section: &'a [u8],
+}
```

Comment:
I'm only seeing half the design here in this PR since I'm not sure how this will be consumed, but could the functionality on this structure be folded directly into `wasmtime::Module` itself? For example there's not really any need to create a duplicate `Vec<FunctionLoc>` when it already lives within `wasmtime::Module`. Additionally this structure may grow somewhat significantly over time to end up looking like a lot of interior pointers into `wasmtime::Module` and otherwise it may not need its own state. 

That being said though your future plans for this structure may motivate a separate type.